docs/topics/api/shelves.rst

```diff
@@ -101,4 +101,19 @@ Current implementation relies on Adzerk to determine which addons are returned a
     :>json string results[].click_url: the url to ping if the sponsored addon's detail page is navigated to.
     :>json string results[].click_data: the data payload to send to ``click_url`` that identifies the sponsored placement clicked on.
     :>json string impression_url: the url to ping when the contents of this sponsored shelf is rendered on screen to the user.
-    :>json string impression_data: the data payload to send to ``impression_url`` that identifies the sponsored placements displayed.
+    :>json string impression_data: the signed data payload to send to ``impression_url`` that identifies the sponsored placements displayed.
+
+
+---------------------------
+Sponsored Shelf Impressions
+---------------------------
+
+.. _sponsored-shelf-impression:
+
+When the sponsored shelf is displayed for the user this endpoint can be used to record the impressions.
+The current implemenation forwards these impression pings to Adzerk.
+
+
+.. http:post:: /api/v4/shelves/sponsored/
+
+    :query string impression_data: the signed data payload that was sent in the :ref:`sponsored shelf <sponsored-shelf>` response.
```

Comment:
We're going to use the exact same endpoint for requesting shelf data, sending impression pings, and sending click pings, and just use the associated data to determine what type of API call it is? I suppose that's fine, but it seems a bit complicated from a caller perspective to me. Also, this data will be sent in a POST. Is it accurate to call this `:query`?

Reply:
copypasta fail 🙄 - should be `/shelves/sponsored/impression/'.  

And I guess it isn't `query` exactly - I'll look into how POST, but non-JSON, data is supposed to be specified.

Reply:
https://github.com/mozilla/addons-server/pull/15687/commits/698b50d3d3d791a7ce33e9782f4aaca82b3009ba